query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java

```diff
@@ -150,10 +150,7 @@ private boolean shouldModifyIndexes(FlagAffectedCommand command, InvocationConte
 
       if (value == null) return cacheMode.isReplicated() || ctx.isOriginLocal();
```

Comment:
I never understood the purpose of the null check in the previous version of this method. 
```indexingMode``` is never null, so the ```return indexingMode.shouldModifyIndexes(command, ctx, distributionManager, rpcManager, key);``` was always executed and the rest of the method is dead code.

Reply:
I think it's because some commands don't have value, e.g. ```ClearCommand```

Reply:
Yes, no value, but I was not referring to the value null check. I meant indexingMode null check.

Reply:
We didn't unriddle this yet. There was some dead code there because indexingMode was never null. ```if (value == null) return cacheMode.isReplicated() || ctx.isOriginLocal();``` is dead code, handling a special case. The ```value``` parameter of that method can be removed too.  Strange. I could not find the reason for this on 9.4.x, 10.1.x, master. So this oddity must be older, if it ever made sense. What do we do with it @gustavonalle ?

Reply:
It was because the IndexingMode got deprecated, and it could be null when not used

Reply:
So since you removed the check, it will simply start to ignore the user setting, and the reason it did not break any test, because there are no test for this legacy setting...Please put it back for now

Reply:
@gustavonalle This is a small issue you have introduced with "ISPN-11241 Deprecate Indexing mode". I merged that PR and did not comment on this aspect although it did look suspect, to expedite things, but would be good to solve it, finally.  We can quickly settle if you follow my reasoning. Field ```IndexModificationStrategy indexingMode``` could never be null (or become null, it's final), starting since its introduction by Sanne with "ISPN-4628 Refactoring: unify LocalQueryInterceptor and QueryInterceptor".  IndexModificationStrategy instances cannot be directly set from config. They are created by logic in method IndexModificationStrategy configuredStrategy(...), based on config, and that method never returns null; worst case it returns ALL. This situation did not change after your PR (that, and this). Or should it have retuned null if the user did not specify an index mode after ISPN-11241 ?

ATM the null check you added for this field with "ISPN-11241 Deprecate Indexing mode" does not seem to have purpose and the code branch in which indexingMode is presumed null is dead code.  And I would like to undo that mistake, since query is loaded with such convoluted things that nobody understands anymore. Of course there are no tests covering the 'legacy' IndexMode == null :)

Given the above reasoning, ``` if (value == null) return cacheMode.isReplicated() || ctx.isOriginLocal();``` is a piece of dead code ATM that handles a corner case that does not really exist here and the commands that do not have a value are handled somewhere else.

Reply:
You are right, I can't remember why I put that null check, maybe because it was causing a NPE, then the code changed, not sure now. In any case, we cannot get rid of the configured index mode, since it's only deprecated, we need to add a test for it

Reply:
or better saying, keep at least 1 test using them, I think probably your WIP would break :)

Reply:
In any case, let's handle it in another JIRA, this is about to removing elasticsearch

Reply:
> You are right, I can't remember why I put that null check, maybe because it was causing a NPE, then the code changed, not sure now. In any case, we cannot get rid of the configured index mode, since it's only deprecated, we need to add a test for it.

It's only deprecated because we did not want to prevent old configs from being parsed, but it is actually ignored. Some indexed modes (like LOCAL) got deprecated gradually and they were automatically handled as their closest equivalent. But after all these removals we came to the point we no longer have any index mode. All eventualy configured index mode values are now effectively handled as if it were ```ALL```.  We do not need tests to show that the index mode config is supported because it isn't really. And writing tests to prove it is ignored is a pointless exercise. I think we're good with what we have now (not saying it's ideal).